script.js

```diff
@@ -3,25 +3,45 @@ document.addEventListener("DOMContentLoaded", () => {
   const userInput = document.getElementById("user-input");
 
   // Add a new line to the output
+  //TODO: handle special characters
   function addLine(text) {
     const line = document.createElement("div");
     line.textContent = text;
     output.appendChild(line);
     output.scrollTop = output.scrollHeight;
   }
 
+  function FileSysObj(isdir, modified, path, children, data){
+    this.isdir = isdir;
+    this.modified = modified; 
+    this.path = path;
+    this.children = children;
+    this.data = data;
+  }
+  
   // File system simulation
-  let fs = {
-    "/": ["home", "bin", "etc"],
-    "/home/": ["user"],
-    "/home/user/": ["documents", "pictures", "notes.txt"],
-    "/home/user/notes.txt": "This is a text file with some notes.",
-    "/home/user/documents/": [],
-    "/home/user/pictures/": [],
-    "/bin/": ["bash", "ls", "echo"],
-    "/etc/": ["hosts", "passwd"],
-  };
+  let fs = {};
+
+  function addFile(fileObj){
+    parent = fileObj.path.slice(0, fileObj.path.slice(0, -2).lastIndexOf("/") + 1);
+    if (parent == "")
+      parent = "/";
+    dirName = fileObj.path.slice(fileObj.path.slice(0, -2).lastIndexOf("/") + 1);
+    fs[fileObj.path] = fileObj;
+    fs[parent].children.push(dirName);
+  }
+  
+  fs["/"] = new FileSysObj(true, new Date(), "/", [], null);
+  
+  addFile(new FileSysObj(false, new Date(), "/test", [], "some data\nhow?") );
+  addFile(new FileSysObj(true, new Date(), "/home/", [], null));
+  addFile(new FileSysObj(true, new Date(), "/dev/", [], null));
+  addFile(new FileSysObj(true, new Date(), "/usr/", [], null));
+  addFile(new FileSysObj(true, new Date(), "/home/salastro/", [], null));
+  addFile(new FileSysObj(false, new Date(), "/home/salastro/file.txt", [], "hello there!\n"));
```

Comment:
Could you make a tree structure that gets converted to function calls instead of this? It would be cleaner and easier to deal with.